### PR TITLE
Get a start on implementing fracbyte functionality.

### PR DESCRIPTION
**Pull request by Lyxal**
[We're entering the fracbyte market](https://github.com/AndrovT/Vycoder).

This PR incorporates @AndrovT's brilliant `vycoder` library into the language. Ported to scala as [Vyncode](https://github.com/Vyxal/Vyncode)

How:

- The `!` flag reads a string of raw bits. This might have to be changed for technical scoring reasons, but who knows.
- The `=` flag reads a program as usual, but prints the bit string.

For the website, the result of compiling vyxal/vyncode as scala.js is used

For the offline interpreter, the jar is used.

### CODE REVIEW

File: static/main.js [hunk 1]
```diff
@@ -627,7 +639,7 @@ function shareOptions(shareType) {
             output = `[Vyxal, ${len} byte${"s".repeat(code.length != 1)}${utfable ? '' : ' (UTF-8)'}](${url})`
             break
         case "post-template":
-            output = `# [Vyxal](https://github.com/Vyxal/Vyxal)${flagAppendage} ${len} byte${"s".repeat(len != 1)}${utfable ? '' : ' (UTF-8)'}
+            output = `#[Vyxal](https://github.com/Vyxal/Vyxal)${flagAppendage} ${len} byte${"s".repeat(len != 1)}${utfable ? '' : ' (UTF-8)'}
```
Comment (Steffan153) [inline static/main.js:642]:
```suggestion
            output = `# [Vyxal](https://github.com/Vyxal/Vyxal)${flagAppendage} ${len} byte${"s".repeat(len != 1)}${utfable ? '' : ' (UTF-8)'}
```

**Comment by GingerIndustries** [comment]
Don't merge this yet!

**Comment by nayakrujul** [comment]
@GingerIndustries ik, I read the chat

**Review by Steffan153 — COMMENTED** [review]
Huh? Vycoder already exists in python, why not just use the python?

**Comment by Steffan153** [comment]
For JS you could use something like https://github.com/flexxui/pscript, or just add an API endpoint to Django and use that

**Comment by Lyxal** [comment]
> For JS you could use something like https://github.com/flexxui/pscript, or just add an API endpoint to Django and use that

The library doesn't seem to support imports nor do conversion of all numbers to big ints (numbers involved in the encoding/decoding will exceed js number limits).

And I'd rather not have an api to the server because it'd be making a lot of requests on a server that's already questionably fragile.

**Comment by Lyxal** [comment]
Having it in scala also means there's less work to do when it eventually gets applied to version 3

**Review by Steffan153 — APPROVED** [review]
Fine, fine, fine. But I don't like the idea of a bloated site.

**Comment by Lyxal** [comment]
so turns out decoding is a little broken. I'll fix that and see if adding some recent answers to the corpus improves anything.

**Comment by Lyxal** [comment]
Decoding now works good online (it was a bug with not resetting class fields when calling Initalise). I'll play around with the corpus later

**Pull request merged**